### PR TITLE
Fix potential API mismatch in BuildsService

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Ran into another mismatch & realized a few methods in BuildsService are probably not future proofed.

### CODE REVIEW

File: app/scripts/services/builds.js [hunk 1]
```diff
@@ -88,8 +88,10 @@ angular.module("openshiftConsole")
         namespace: build.metadata.namespace
       };
       var canceledBuild = angular.copy(build);
+      var rgv = APIService.objectToResourceGroupVersion(canceledBuild);
       canceledBuild.status.cancelled = true;
-      return DataService.update(buildsVersion, canceledBuild.metadata.name, canceledBuild, context).then(function() {
```
Comment (spadgett) [inline app/scripts/services/builds.js]: Should be able to remove `buildsVersion` var now

Comment (benjaminapetersen) [reply app/scripts/services/builds.js]: Not sure why jshint didn't catch that when I did my build, just removed.

File: app/scripts/services/builds.js [hunk 2]
```diff
@@ -108,16 +109,18 @@ angular.module("openshiftConsole")
     var cloneBuild = function(originalBuild, buildConfigName) {
       var buildType = isJenkinsPipelineStrategy(originalBuild) ? 'pipeline' : 'build';
       var originalDisplayName = getBuildDisplayName(originalBuild, buildConfigName);
+
       var req = {
         kind: "BuildRequest",
-        apiVersion: "v1",
+        apiVersion: originalBuild.apiVersion,
```
Comment (spadgett) [inline app/scripts/services/builds.js]: This should be the api version for `buildsCloneVersion`, which might not match `originalBuild.apiVersion`

Comment (benjaminapetersen) [reply app/scripts/services/builds.js]: Ah, so `APIService.toAPIVersion(buildsCloneVersion)`, ill update.

Comment (benjaminapetersen) [reply app/scripts/services/builds.js]: I bet I'm gonna be doing a # of bug fixes around this.  Now it's getting fiddly.

Comment (benjaminapetersen) [reply app/scripts/services/builds.js]: updated

**Pull request merged**